tests/cpp/test_alias.cpp

```diff
@@ -1659,4 +1660,39 @@ TEST_F(AliasTest, SliceOfExpandedBroadcast) {
       executor_cache.fusion(), out_tensors, {in_tensor}, __LINE__, __FILE__);
 }
 
+TEST_F(AliasTest, BroadcastInDimNoRedundantSet) {
+  // Test that broadcast with no actual broadcasting does not introduce
+  // a redundant Set operation
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  TensorView* in = makeContigConcreteTensor({2, 3});
+  fusion->addInput(in);
+  
+  // Call broadcast with all dims marked as non-broadcast
+  // This should not introduce a Set operation
+  std::vector<bool> is_broadcast_dim = {false, false};
+  TensorView* out = broadcast(in, is_broadcast_dim);
```

Comment:
The fusion is empty, so the verification seems silly.
```suggestion
  TensorView* maybe_bcast = broadcast(in, is_broadcast_dim);
  TensorView* out = abs(maybe_bcast);
```

Or just check: 
`EXPECT_EQ(fusion->exprs(), 0);`

Reply:
@copilot, please implement the change according to feedback from Ryan.